extensions/vscode/src/commands/install-cli.ts

```diff
@@ -0,0 +1,48 @@
+const cp = require("child_process");
+
+import { window, ProgressOptions } from "vscode";
+
+/**
+ * Installs Dart Frog CLI in the user's system if not already installed.
+ */
+export const installCLI = async (): Promise<void> => {
+  if (!hasDartFrogCliInstalled()) {
+    const options: ProgressOptions = {
+      location: 15,
+      title: "Installing Dart Frog CLI...",
+    };
+    window.withProgress(options, installDartFrogCliVersion);
+  }
+};
+
+/**
+ * Whether the user has Dart Frog CLI installed in their system.
+ *
+ * @returns {boolean} True if the user has Dart Frog CLI installed in their
+ * system, false otherwise.
+ */
+function hasDartFrogCliInstalled(): boolean {
+  try {
+    cp.execSync(`dart_frog --version`);
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+/**
+ * Installs Dart Frog CLI from pub.dev.
+ *
+ * @returns {Promise<void>} A promise that resolves when the installation is
+ * complete.
+ */
+async function installDartFrogCliVersion(): Promise<void> {
+  await cp.exec(
+    `dart pub global activate dart_frog_cli`,
```

Comment:
We should ensure this installs a version of the CLI compatible with the extension.

Reply:
I agree, I believe adding this functionality will require some additional logic that can pollute this PR. I'm thinking on adding this suggested functionality (ensuring a compatible version of Dart Frog CLI is installed) as a separate PR. What do you think?

Reply:
I agree, I'm planning to do this on a different pull request to avoid polluting this with the functionality you mentioned. How does this sound to you?

Reply:
I'll be tackling this in another Pull Request.